android-json-form-wizard/src/main/java/com/vijay/jsonwizard/interactors/JsonFormInteractor.java

```diff
@@ -173,16 +180,42 @@ public void fetchFields(List<View> viewsFromJson, String stepName, JsonFormFragm
 
             for (int i = 0; i < fields.length(); i++) {
                 JSONObject childJson = fields.getJSONObject(i);
-                fetchViews(viewsFromJson, stepName, formFragment, childJson.getString(JsonFormConstants.TYPE), childJson,
+                List<View> views = fetchViews(viewsFromJson, stepName, formFragment, childJson.getString(TYPE), childJson,
                         listener, popup);
+                if (childJson.get(TYPE).equals(REPEATING_GROUP)) {
+                    for (int j = i + 1; j < fields.length(); j++) {
```

Comment:
Why not use a `while` loop here instead of having a nested `if` statement with an `else` that does not contain logic?